crescent/commands/errors.py

```diff
@@ -0,0 +1,59 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Callable, Generic, Protocol, Type, TypeVar
+
+from attrs import define
+
+from crescent.internal.meta_struct import MetaStruct
+
+if TYPE_CHECKING:
+    from crescent.context import Context
+    from crescent.internal.app_command import AppCommandMeta, CommandCallback
+    from crescent.typedefs import CommandOptionsT
+
+
+ERROR = TypeVar("ERROR", bound=Exception, contravariant=True)
+
+
+class ErrorHandlerProto(Protocol, Generic[ERROR]):
+    async def __call__(
+        self,
+        /,
+        *,
+        exc: ERROR,
+        ctx: Context,
+        command: MetaStruct[CommandCallback, AppCommandMeta],
+        options: CommandOptionsT,
+    ) -> None:
+        ...
+
+
+@define
+class ErrorHandlerMeta:
```

Comment:
What does this do?

Reply:
Though I guess you can do `MetaStruct[ErrorHandlerProto, Any]`